### PR TITLE
chore: #594 backend cleanup alembic remnants

### PR DESCRIPTION
**Pull request by ianliuwk1019**
This PR #594 contains changes for:

* Remove server/backend/api/alembic migration folder.
* Remove server/backend/tests folder related to sqllite.
* Update some imports for tests.
* Remove sqllite tests from unit-test workflow.
* Update readme.
* Use lib "**pytest-dotenv**" to load our '**local-dev.env**' with **pytest.ini** to unify backend just need **one** .env file without duplicating the values in other places.
* Modify docker-compose.yml to "extends" from **docker-base-service.yml** so FAM can have separate **docker-compose-testcontainer.yml** also extends from base services but leaving **api service** out from the use in testcontainer workflow, so no need to spin out redundant 'api' container. This saves about 1/3 of running unittest workflwo time.
* Adjust ".github/workflows/reuseable_api_unittests.yml" to work from "**server/backend**" "**working-directory**".

### CODE REVIEW

**Comment by basilv** [comment]
The PR should probably be named a 'chore' rather than a 'fix' in terms of updating version number and release notes, but that's a nitpicky point.

**Comment by ConradBoydElliottGustafson** [comment]
> The PR should probably be named a 'chore' rather than a 'fix' in terms of updating version number and release notes, but that's a nitpicky point.

I think anything other than "fix" or "feat" does not bump the version number. Pretty sure we want a version number bump for this? Am I wrong?

**Comment by basilv** [comment]
Hard to say, if we are only changing unit test setup, and there's nothing different with the actual deployed app then no need for a deploy or version change in theory. But lots of files have changed...

In practice, I'm fine with a version change for this, I just was trying to fit this within the release-notes model.

**Comment by ianliuwk1019** [comment]
> Hard to say, if we are only changing unit test setup, and there's nothing different with the actual deployed app then no need for a deploy or version change in theory. But lots of files have changed...
> 
> In practice, I'm fine with a version change for this, I just was trying to fit this within the release-notes model.

Thanks Basil. I changed to "chore".

File: docker-compose.yml [hunk 1]
```diff
@@ -2,58 +2,17 @@
...
-      - FLYWAY_PLACEHOLDERS_client_id_dev_forest_client_oidc_client=xxqiHFmwG8j1cVAz7NgtknaZOg
-      - FLYWAY_PLACEHOLDERS_client_id_test_forest_client_oidc_client=dm5Xkmomnq0gbwBiXiN5LgAnah
-      - FLYWAY_PLACEHOLDERS_client_id_prod_forest_client_oidc_client=KdnD2eGS3Zcx494p04yMFhDwSe
+    extends:
+      file: docker-base-services.yml
+      service: fam-flyway
     depends_on:
       fam-database:
         condition: service_healthy
-    networks:
```
Comment (ConradBoydElliottGustafson) [inline docker-compose.yml:55]: Do we not need the network fam anymore?

Comment (ConradBoydElliottGustafson) [reply docker-compose.yml:55]: I tested locally, seems to work fine without the network explicitly defined.

**Review by ConradBoydElliottGustafson — APPROVED** [review]
I rebased the branch for you.

**Pull request merged**